scripts/bottleneck_detector.py

```
@@ -0,0 +1,583 @@
+            
+        sorted_data = sorted(data)
+        index = (percentile / 100) * (len(sorted_data) - 1)
+        
+        if index.is_integer():
+            return sorted_data[int(index)]
+        else:
+            lower = sorted_data[int(index)]
+            upper = sorted_data[min(int(index) + 1, len(sorted_data) - 1)]
+            return lower + (upper - lower) * (index - int(index))
+            
+    def detect_long_running_steps(self, duration_threshold_percentile: float = 95,
```

Ruff (0.14.4)

Unused method argument: `duration_threshold_percentile`

(ARG002)